### PR TITLE
Ds/advec1d

### PR DESCRIPTION
**Pull request by dsteinmo**
hey bud, here's that forward Euler code you were asking for.

### CODE REVIEW

**Comment by dsteinmo** [comment]
TODO: Update the headers zip in the S3 bucket to include the boost lexical casts stuff (to fix the appveyor build). Edit: Nevermind, going to just use stringstreams instead of lexical cast.

File: src/Nodes1DProvisioner.cpp [hunk 1]
```diff
@@ -80,6 +86,23 @@ void Nodes1DProvisioner::buildNodes() {
...
+ */
+void Nodes1DProvisioner::buildNormals() {
+    matrix_type & nxref = *nx;
+
+    index_type count = 0;
+    for (index_type k=0; k < NumElements; k++) {
+        for (index_type f=0; f < NumFaces*NumFacePoints; f++) {
+            nxref(f,k) = pow(-1., count+1);
+            count++;
+        }
```
Comment (k7miller) [inline src/Nodes1DProvisioner.cpp]: Hey bud, I know you love the pow function ;-), but I would implement this as

real_type mult = -1;

and then inside the function:

nxref(f, k) = count;
count *= -1;

File: src/main.cpp [hunk 2]
```diff
@@ -11,60 +11,150 @@
...
-  MeshManager * manager = new MeshManager();
-  MeshManager & mgr = *manager;
-  mgr.readVertices("input/2box.V");
-  mgr.readElements("input/2box.E2V");
+		// Calculate Righ-hand side at current time-step.
+		computeRHS(u, c, nodes1DProvisioner, RHS);
 
-  mgr.partitionMesh(2);
+		// Forward Euler time-step for now, to be replaced.
+		u = u + dt*RHS;
```
Comment (k7miller) [inline src/main.cpp]: u += dt*RHS may be more efficient here.

**Review by k7miller — COMMENTED** [review]
These changes look good to me buddy. I only had two minor comments. Nice job!

File: src/Nodes1DProvisioner.cpp (hunk 1 above)
Comment (dsteinmo) [reply src/Nodes1DProvisioner.cpp]: The pow function is the greatest, but I will abide by your advice, lol.

**Pull request merged**